contracts/contracts/RequestManager.sol

```diff
@@ -299,46 +307,74 @@ contract RequestManager is Ownable {
+        // 2) DepositReceiver, the claimer is the address that withdrew the deposit with another claim
+        // 3) Claim properties, claim terminated and claimer has the highest stake
         address filler = resolutionRegistry.fillers(fillHash);
         if(filler == address(0)) {
             filler = depositReceiver;
         }
-
         if (filler == address(0)) {
-            // Claim resolution via claim properties
+            // Claim resolution via 3)
             require(block.timestamp >= claim.termination, "Claim period not finished");
-            claimReceiver = claim.claimerStake > claim.challengerStake ? claim.claimer : claim.challenger;
-        } else if (filler != claim.claimer) {
-            // Claim resolution via 1) or 2) but claim is invalid (challenger wins challenge)
-            claimReceiver = claim.challenger;
-        } else {
-            // Claim resolution via 1) or 2) and claim is valid (claimer wins challenge)
-            claimReceiver = claim.claimer;
+            claimValid = claimerStake > challengerStakeTotal;
+        }
+        else {
+            // Claim resolution via 1) or 2)
+            claimValid = filler == claim.claimer;
         }
 
-        claim.withdrawn = true;
-        request.activeClaims -= 1;
+        // Calculate withdraw scheme for claim stakes
+        uint256 ethToTransfer;
+        address claimReceiver;
 
-        if (depositReceiver == address(0) && claimReceiver == claim.claimer) {
-            withdrawDeposit(request, claim, claimReceiver);
+        if (claimValid) {
+            // If claim is valid, all stakes go to the claimer
+            ethToTransfer = claimerStake + challengerStakeTotal;
+            claimReceiver = claim.claimer;
+        }
+        else if (challengerStakeTotal > 0) {
+            // If claim is invalid, partial withdrawal by the sender
+            ethToTransfer = 2 * claim.challengersStakes[msg.sender];
+            claimReceiver = msg.sender;
+
+            require(ethToTransfer > 0, "Challenger has nothing to withdraw");
+            //Re-entrancy protection
+            claim.challengersStakes[msg.sender] = 0;
+        }
+        else {
+            // The unlikely event is possible that a false claim has no challenger
+            // If it is known that the claim is false then the claim stake goes to the platform
+            ethToTransfer = claimerStake;
+            claimReceiver = owner();
+        }
+
+        // If the challenger wins and is the last challenger
+        // he gets either the excess stakes if claimer was winning
+        // or twice his stake minus the difference of the claimer and challenger stakes
+        if (msg.sender == claim.lastChallenger) {
+            if (claimerStake > challengerStakeTotal) {
+                ethToTransfer += (claimerStake - challengerStakeTotal);
+            }
+            else {
+                ethToTransfer -= (challengerStakeTotal - claimerStake);
+            }
```

Comment:
The comment above this block is slightly wrong: in one case the challenger gets "the excess", but in the other he gets the full amount (2*stake minus the difference).
In order to be consistent, I'd suggest something like
```
        // If the challenger wins and is the last challenger, he gets either
        // twice his stake plus the excess stake if claimer was winning, or
        // twice his stake minus the difference of the claimer and challenger stakes
```